implementations/rust/ockam/ockam_api/src/nodes/service/secure_channel.rs

```diff
@@ -271,8 +271,10 @@ impl NodeManager {
     pub(super) async fn delete_secure_channel_listener_impl(
         &mut self,
         addr: &Address,
-    ) -> Option<SecureChannelListenerInfo> {
+        ctx: &Context
+    ) -> Option<SecureChannelListenerInfo>  {
         debug!("deleting secure channel listener: {addr}");
+        let _ = self.secure_channels.stop_secure_channel(ctx, addr).await;
```

Comment:
This is good, but I think we could go a step further. The inner logic of the `stop_secure_channel` function was recently refactored and it's now the same as calling `ctx.stop_worker` (it used to be a more complex piece of code).

So, my proposal would be:
- Remove the `stop_secure_channel`
- Use `ctx.stop_worker(address.clone()).await` in its place